internal/services/manageddevopspools/managed_devops_pool_resource.go

```diff
@@ -0,0 +1,746 @@
+						}
+					}
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ManagedDevOpsPoolResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ManagedDevOpsPools.PoolsClient
+
+			id, err := pools.ParsePoolID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+			return nil
+		},
+	}
+}
+
+func (ManagedDevOpsPoolResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return pools.ValidatePoolID
+}
+
+func (ManagedDevOpsPoolResource) CustomizeDiff() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model ManagedDevOpsPoolModel
+			if err := metadata.DecodeDiff(&model); err != nil {
+				return fmt.Errorf("DecodeDiff: %+v", err)
+			}
+
+			// Validate vmss_fabric_profile images with proper handling of computed/unknown values
+			if err := validateVmssFabricProfileImages(metadata, model.VmssFabricProfile); err != nil {
+				return err
+			}
+
+			for _, orgProfile := range model.AzureDevOpsOrganizationProfile {
+				for _, permProfile := range orgProfile.PermissionProfile {
+					if permProfile.Kind != string(pools.AzureDevOpsPermissionTypeSpecificAccounts) {
+						if len(permProfile.AdministratorAccounts) > 0 {
+							return fmt.Errorf("`administrator_account` block is not required when `permission_profile` kind is `%s`", permProfile.Kind)
+						}
+					}
+				}
+			}
+
+			// Validate agent counts don't exceed maximum_concurrency
+			maxConcurrency := model.MaximumConcurrency
+
+			// Validate stateful agent profile schedules
+			for _, statefulProfile := range model.StatefulAgentProfile {
+				for _, manualProfile := range statefulProfile.ManualResourcePredictionsProfile {
+					if err := validateManualProfileAgentCounts("stateful_agent_profile", manualProfile, maxConcurrency); err != nil {
+						return err
+					}
+				}
+			}
+
+			// Validate stateless agent profile schedules
+			for _, statelessProfile := range model.StatelessAgentProfile {
+				for _, manualProfile := range statelessProfile.ManualResourcePredictionsProfile {
+					if err := validateManualProfileAgentCounts("stateless_agent_profile", manualProfile, maxConcurrency); err != nil {
+						return err
+					}
+				}
+			}
+
+			return nil
+		},
+		Timeout: 5 * time.Minute,
+	}
+}
+
+func validateVmssFabricProfileImages(metadata sdk.ResourceMetaData, vmssFabricProfiles []VmssFabricProfileModel) error {
+	rawConfig := metadata.ResourceDiff.GetRawConfig().AsValueMap()
+
+	vmssFabricValue, exists := rawConfig["vmss_fabric_profile"]
+	if !exists || vmssFabricValue.IsNull() {
+		return nil
+	}
+
+	vmssFabricSlice := vmssFabricValue.AsValueSlice()
```

Comment:
This will panic if the `vmssFabricValue` is unknown. Instead, you can call `vmssFabricValue.IsWhollyKnown` and return early if it is false. Otherwise if it is known, you can just do the validation based on `vmssFabricProfiles`.